server/cloudformation/ecs.go

```diff
@@ -174,6 +180,235 @@ func (p *ECSServiceResource) delete(ctx context.Context, service, cluster *strin
 	return nil
 }
 
+type PortMapping struct {
+	ContainerPort *customresources.IntValue
+	HostPort      *customresources.IntValue
+}
+
+type Ulimit struct {
+	Name      *string
+	HardLimit *customresources.IntValue
+	SoftLimit *customresources.IntValue
+}
+
+type ContainerDefinition struct {
+	Name             *string
+	Command          []*string
+	Cpu              *customresources.IntValue
+	Image            *string
+	Essential        *string
+	Memory           *customresources.IntValue
+	PortMappings     []PortMapping
+	DockerLabels     map[string]*string
+	Ulimits          []Ulimit
+	Environment      []string
+	LogConfiguration *ecs.LogConfiguration
+}
+
+// TaskDefinitionProperties are properties passed to the
+// Custom::ECSTaskDefinition custom resource.
+type ECSTaskDefinitionProperties struct {
+	Family               *string
+	TaskRoleArn          *string
+	ContainerDefinitions []ContainerDefinition
+}
+
+// ECSTaskDefinitionResource is a custom resource that provisions ECS task
+// definitions.
+type ECSTaskDefinitionResource struct {
+	ecs              ecsClient
+	environmentStore environmentStore
+}
+
+func (p *ECSTaskDefinitionResource) Properties() interface{} {
+	return &ECSTaskDefinitionProperties{}
+}
+
+func (p *ECSTaskDefinitionResource) Provision(ctx context.Context, req customresources.Request) (string, interface{}, error) {
+	properties := req.ResourceProperties.(*ECSTaskDefinitionProperties)
+
+	switch req.RequestType {
+	case customresources.Create:
+		id, err := p.register(properties, req.Hash())
+		return id, nil, err
+	case customresources.Delete:
+		id := req.PhysicalResourceId
+		err := p.delete(id)
+		return id, nil, err
+	case customresources.Update:
+		id, err := p.register(properties, req.Hash())
+		return id, nil, err
+	default:
+		return "", nil, fmt.Errorf("%s is not supported", req.RequestType)
+	}
+}
+
+func (p *ECSTaskDefinitionResource) mergedEnvironment(ids ...string) ([]*ecs.KeyValuePair, error) {
```

Comment:
this seems more like `resolvedEnvironment` to me, you're resolving a list of keys into the actual values vs. merging anything